extension/parquet/parquet_extension.cpp

```diff
@@ -116,6 +116,7 @@ struct ParquetWriteBindData : public TableFunctionData {
 	vector<string> column_names;
 	duckdb_parquet::format::CompressionCodec::type codec = duckdb_parquet::format::CompressionCodec::SNAPPY;
 	idx_t row_group_size = RowGroup::ROW_GROUP_SIZE;
+	idx_t row_group_size_bytes = 134217728; // 128MB (1 << 27)
```

Comment:
Maybe it makes sense to default this to e.g. 1KB per row? That way the default scales with the user-specified `ROW_GROUP_SIZE`. For our default row group size that would be 128MB. But that would mean that if the user increases the row group size manually (to e.g. 1 million rows) the `row_group_size_bytes` would automatically scale with the user-specified row group size.

Reply:
So just to clarify, if a user sets only the `ROW_GROUP_SIZE` to N, then we flush the row group when we either exceed N rows, or N kb.

If a user sets both `ROW_GROUP_SIZE` to N and `ROW_GROUP_SIZE_BYTES` to M, we flush the row group when we exceed N rows or M bytes.

If a user sets only `ROW_GROUP_SIZE_BYTES` to M, we flush the row group when we exceed the default 122880 rows, or M bytes.

So `ROW_GROUP_SIZE` has an effect on `ROW_GROUP_SIZE_BYTES`, but not the other way around.

Is that what you mean? That sounds reasonable

Reply:
Yes, exactly, which in the current implementation means that when `ROW_GROUP_SIZE_BYTES` is not set it defaults to `ROW_GROUP_SIZE * 1KB`.